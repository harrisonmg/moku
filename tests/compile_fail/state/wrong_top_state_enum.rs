use moku::*;

#[state_machine]
mod blinky {
    use moku::*;

    #[machine_module]
    mod state_machine {}

    #[derive(Debug, Clone, Copy)]
    enum BlonkyState {}
    impl StateEnum for BlonkyState {}

    struct Top {}
    impl TopState<BlonkyState> for Top {}
}

fn main() {}
