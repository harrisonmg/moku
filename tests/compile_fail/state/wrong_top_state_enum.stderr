error: implementations of moku::TopState in this module must use only BlinkyState as the generic
  --> tests/compile_fail/state/wrong_top_state_enum.rs:15:5
   |
15 |     impl TopState<BlonkyState> for Top {}
   |     ^^^^
